### PR TITLE
Add syllabus data structures

### PR DESCRIPTION
**Pull request by korgan00**
## Changes

Fix https://github.com/Qiskit/platypus/issues/851

## Implementation details

- Add the data structures in a ts file.
- Use the data structures in all the components.
- Adapt the behavior of the components to the new structures.

## How to read this PR

`/syllabus/123-abc`
`/syllabus/create`
`/account#Classroom`

### CODE REVIEW

File: frontend/ts/syllabus.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  title: string,
+  description: string,
+  unitList: Unit[],
+}
+
+export interface Unit {
+  name: string,
+  url: string,
+  id: string
+}
```
Comment (Tansito) [inline frontend/ts/syllabus.ts:27]: The eslint probably will need a newline at the end of the file.

File: frontend/vue/components/Syllabus/SyllabusForm.vue [hunk 2]
```diff
@@ -90,36 +100,81 @@ export default defineComponent({
...
+        institution: 'UCM',
+        officeHours: '10:00 to 13:00',
+        classHours: '15:00 to 18:00',
+        email: 'none@none.never',
+        courseList: [
+          {
+            title: 'Week 1',
+            description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin vehicula tellus non ligula hendrerit interdum. Suspendisse sit amet erat vitae urna mattis sodales. Nullam consequat sagittis tellus. In et justo ex. Suspendisse tempor auctor blandit. Sed vel est eu felis vehicula varius id non ante. Morbi lacinia dolor ac nunc malesuada, dictum imperdiet ligula pellentesque.',
+            unitList: [
+              {
```
Comment (Tansito) [inline frontend/vue/components/Syllabus/SyllabusForm.vue:117]: Similar to what I mention previously. The backend initially will return an array of UUIDs like:

```js
['bdb1d662-e0f6-428a-8830-befe6b47f320', '7c765036-aebc-11ec-b909-0242ac120002']
```

File: frontend/ts/syllabus.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  email: string,
+  courseList: SyllabusCourse[]
+}
+
+export type SyllabusCode = string
+
+export interface SyllabusCourse {
+  title: string,
+  description: string,
+  unitList: Unit[],
```
[the remark this answers is not in the document]
Comment (korgan00) [reply frontend/ts/syllabus.ts:20]: I will change because, finally i only use the uuid

File: frontend/vue/components/Syllabus/SyllabusForm.vue [hunk 4]
```diff
@@ -90,36 +100,70 @@ export default defineComponent({
           location: 'create-syllabus'
         }
       },
-      modules: [{
-        id: 1
-      }],
-      modulesCounter: 1
+      sections: [] as Section[],
```
Comment (Tansito) [inline frontend/vue/components/Syllabus/SyllabusForm.vue:103]: Are these sections something that we can remove?

Comment (Tansito) [inline frontend/vue/components/Syllabus/SyllabusForm.vue:103]: Correct me if I'm wrong @korgan00 , these sections are not related with the **syllabus** entity, aren't they?

File: frontend/vue/components/Syllabus/SyllabusFormCourseInfo.vue [hunk 5]
```diff
@@ -98,37 +105,56 @@ import 'carbon-web-components/es/components/tabs/tabs.js'
...
+        name: (this.$refs.nameInput as BXInput).value,
+        instructor: (this.$refs.instructorInput as BXInput).value,
+        location: (this.$refs.locationInput as BXInput).value,
+        institution: (this.$refs.institutionInput as BXInput).value,
+        officeHours: (this.$refs.officeHoursInput as BXInput).value,
+        classHours: (this.$refs.classHoursInput as BXInput).value,
+        email: (this.$refs.emailInput as BXInput).value
+      }
+
+      const mergedObject = {
```
Comment (Tansito) [inline frontend/vue/components/Syllabus/SyllabusFormCourseInfo.vue:152]: I don't know if this info can simplify your logic but for the update in backend is more than enough that you send me the fields that you want to modify. You don't need to pass the entire syllabus (just to mention).

**Review by Tansito — APPROVED** [review]

File: frontend/vue/components/Syllabus/SyllabusForm.vue (hunk 4 above)
Comment (korgan00) [reply frontend/vue/components/Syllabus/SyllabusForm.vue:103]: In fact, is not used anywhere. Probably I forgot to remove it, good catch.

File: frontend/vue/components/Syllabus/SyllabusFormCourseInfo.vue (hunk 5 above)
Comment (korgan00) [reply frontend/vue/components/Syllabus/SyllabusFormCourseInfo.vue:152]: Per offline conversations. This is resolved :)
